CMakeLists.txt

```diff
@@ -64,6 +64,11 @@ if(USE_MKL_IF_AVAILABLE AND (NOT APPLE) AND (NOT MSVC) AND (CMAKE_HOST_SYSTEM_PR
 else()
   option(USE_MKLDNN "Build with MKL-DNN support" OFF)
 endif()
+if ((CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64") AND (NOT CMAKE_CROSSCOMPILING))
+  option(USE_INTGEMM          "Build with x86 intgemm library for low-precision multiplication" ON)
+else()
+  option(USE_INTGEMM          "Build with x86 intgemm library for low-precision multiplication" OFF)
+endif()
```

Comment:
If `USE_INTGEMM` is only supported on x86, you should use `cmake_dependent_option`. If it's supported but not recommended on other architectures, your current approach is fine, though you may want to update the docstring to clarify.

Reply:
8aaa23c5b1f1b3dbef78be229abdc83dbe96216d